### PR TITLE
feat(ingest): extract SageMaker metrics, hyperparameters, and external URLs

### PR DESCRIPTION
**Pull request by kevinhu**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!

File: metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py [hunk 1]
```diff
@@ -255,14 +255,19 @@ class JobDirection(Enum):
...
     Intermediate representation of a job's related models. Subsequently used by the SageMaker jobs ingestion framework.
     """
 
     job_urn: str
     job_direction: JobDirection
+    hyperparameters: Dict[str, Any] = field(default_factory=dict)
+    metrics: Dict[str, Any] = field(default_factory=dict)
+
+    def __hash__(self):
+        return hash((self.job_urn, self.job_direction))
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py:270]: this seems a bit odd to me?

is there a reason we can't use `frozendict`? it's a hashable type so we can continue using `frozen=True`

Comment (kevinhu) [reply metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py:270]: I considered that, but in the end we're only looking for unique jobs by direction and name – if two jobs have different metrics/hyperparams then they must be distinct anyway

**Review by hsheth2 — APPROVED** [review]

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**